libres/lib/analysis/update.cpp

```diff
@@ -107,12 +107,37 @@ class update_data_type {
     matrix_type *D;
     matrix_type *R;
     matrix_type *A;
-    const bool_vector_type *obs_mask;
+    const std::vector<bool> obs_mask;
     std::vector<std::pair<matrix_type *, std::shared_ptr<RowScaling>>>
         A_with_rowscaling;
     bool has_observations = false;
 };
 
+int_vector_type *
```

Comment:
I would put these in an anonymous namespace - or mark them as `static`. Furthermore I would suggest removing the `bool_vector_` part of the function name, since this is part of an effort to completely remove `bool_vector_type` - i.e. something like:
```C++
namespace {
int_vector_type * alloc_active_list(const std::vector<bool>&ens_mask) {
    ...
}
}
```

Reply:
Makes sense :+1: